engines/agi/text.cpp

```diff
@@ -242,18 +242,20 @@ void TextMgr::display(int16 textNr, int16 textRow, int16 textColumn) {
 	char *processedTextPtr   = nullptr;
 
 	charPos_Push();
-	charPos_Set(textRow, textColumn);
```

Comment:
This is a change to non-RTL behavior. Removing this line creates a codepath where charPos_set() does not get called for non-RTL. 

You didn't remove this line in the first version of your PR, so this is more than the requested refactor change. We need to ensure that we are not altering any non-RTL behavior.

Reply:
I went ahead and restored this line and removed it from the else block so it is easier to see that the original behavior hasn't changed.